gef.py

```diff
@@ -8856,22 +8910,33 @@ def do_invoke(self, argv: List[str]) -> None:
         headers = ["Start", "End", "Offset", "Perm", "Path"]
         gef_print(Color.colorify("{:<{w}s}{:<{w}s}{:<{w}s}{:<4s} {:s}".format(*headers, w=gef.arch.ptrsize*2+3), color))
 
+        last_printed_filter = None
+
         for entry in vmmap:
+            names_filter = [f"name = '{x}'" for x in names if x in entry.path]
+            addrs_filter = [f"addr = " + self.format_addr_filter(arg, addr) for arg, addr in addrs.items()
+                if addr >= entry.page_start and addr < entry.page_end]
+            filters = names_filter + addrs_filter
+            filter_content = f"[{' & '.join(filters)}]"
+
             if not argv:
                 self.print_entry(entry)
-                continue
-            if argv[0] in entry.path:
+
+            elif any(filters):
+                if filter_content != last_printed_filter:
+                    gef_print() # skip a line between different filters
+                    gef_print(Color.greenify(filter_content))
+                    last_printed_filter = filter_content
                 self.print_entry(entry)
-            elif self.is_integer(argv[0]):
-                addr = int(argv[0], 0)
-                if addr >= entry.page_start and addr < entry.page_end:
-                    self.print_entry(entry)
-            else:
-                addr = safe_parse_and_eval(argv[0])
-                if addr is not None and addr >= entry.page_start and addr < entry.page_end:
-                    self.print_entry(entry)
+
+        gef_print()
         return
 
+    def format_addr_filter(self, arg, addr):
+        if self.is_integer(arg):
+            return f"`{arg}`"
+        return f"`{arg}` ({addr:#x})"
```

Comment:
This is not a complex function, a one liner is enough
```suggestion
        return f"`{arg}`" if self.is_integer(arg) else f"`{arg}` ({addr:#x})"
```